### PR TITLE
fix(build): patch ngx_lua-1.21.4_07-print-body-double-free bug fix

### PR DESCRIPTION
**Pull request by oowl**
### Summary
This is the backport Openresty patch for https://github.com/openresty/lua-nginx-module/pull/2115

Related: https://github.com/openresty/lua-nginx-module/issues/2184

### Checklist

- [x] The Pull Request has tests
- [ ] There's an entry in the CHANGELOG
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

FTI-5051

### CODE REVIEW

File: t/04-patch/01-ngx-buf-double-free.t [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            ngx.print(str)
+            ngx.flush()
+            ngx.print("small chunk")
+            ngx.flush()
+        }
+        body_filter_by_lua_block {local dummy=1}
+    }
+--- request
+GET /t
+--- response_body_like: small chunk
```
Comment (chronolaw) [inline t/04-patch/01-ngx-buf-double-free.t:24]: Need a "\r" at the end of file.

**Review by ms2008 — COMMENTED** [review]
The patch file should start with `ngx_lua-0.10.21`?

**Review by chronolaw — APPROVED** [review]

**Comment by chronolaw** [comment]
> The patch file should start with `ngx_lua-0.10.21`?

Yes, in OpenResty 1.21.41 ngx_lua is 0.10.21

**Review by bungle — APPROVED** [review]

**Comment by bungle** [comment]
I tested it locally on mac and it works.

**Pull request merged**